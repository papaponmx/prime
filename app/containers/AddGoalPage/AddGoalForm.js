import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm, reducer } from 'redux-form';
import { connect } from 'react-redux';
import { compose } from 'redux';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import { addGoal } from '../GoalsList/actions';
import renderField from './renderField';
import saga from './saga';

const required = value => (value ? undefined : 'Required');
const maxLength = value =>
  value && value.length > 100 ? `Must be ${100} characters or less` : undefined;
const minValue = value =>
  value && value < 3 ? `Must be at least ${3}` : undefined;

/* eslint-disable react/prefer-stateless-function */
class AddGoalForm extends Component {
  render() {
    const { handleSubmit, pristine, reset, submitting } = this.props;
    return (
      <form onSubmit={handleSubmit}>
        <Field
          name="goal"
          type="text"
          component={renderField}
          label="Goal"
          validate={[required, maxLength, minValue]}
        />
        <Field
          name="dueDate"
          type="date"
          component={renderField}
          label="Due date"
        />
        <div>
          <button type="submit" disabled={submitting}>
            Submit
          </button>
          <button
            type="button"
            disabled={pristine || submitting}
            onClick={reset}
          >
            Clear Values
          </button>
        </div>
      </form>
    );
  }
}

// TODO: Add a select with Roles
// TODO: Styles, here.

AddGoalForm.propTypes = {
  handleSubmit: PropTypes.func,
  pristine: PropTypes.any,
  reset: PropTypes.func,
  submitting: PropTypes.any,
};

const mapDispatchToProps = dispatch => ({
  onSubmit: values => {
    dispatch(addGoal(values));
  },
});

const withForm = reduxForm({
  form: 'AddGoalForm',
});

const withConnect = connect(
  null,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'form', reducer });
const withSaga = injectSaga({ key: 'addGoal', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
  withForm,
)(AddGoalForm);
