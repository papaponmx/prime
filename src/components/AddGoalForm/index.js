import React, { Component } from 'react'
import { Field, reduxForm, reset} from 'redux-form'
import { connect } from 'react-redux';
import Button from 'material-ui/Button';

import { addGoal }from '../../actions';
// import { renderSelect } from '../form-utils';

class AddGoalForm extends Component {
  render() {
    // TODO: Rethink goals-roles graph;
    // Should be something like this:
    // [
    //   {
    //     GoalName:
    //     roleId:
    //     dueDate:
    //
    //   }
    // ]

    return (
      <form className="flex-col-start" onSubmit={this.props.handleSubmit}>
      <label id="add-goal-input">New goal</label>
      <Field
      name="goal"
      component="input"
      id="add-goal-input"
      />
      <Field
      name=''
      component='input'
      id='goal-due-date'
      type='date'
      />

      {/* TODO: Add meta-data for goals, like: status = completed, dueDate, priority, etc. */}
      <Button type="submit" className="cta">Add goal</Button>
      </form>
    );
  }
}

// TODO: Create added goal notification success or shit.

const mapStateToProps= state => ({
  roles: state.roles.list,
});


const mapDispatchToProps = (dispatch) => {
  return {
    onSubmit: (values) => {
      dispatch(addGoal(values));
      dispatch(reset('AddGoalForm'));
    },
  }
}

AddGoalForm = reduxForm({
  form: 'AddGoalForm',
})(AddGoalForm);

export default AddGoalForm = connect(mapStateToProps, mapDispatchToProps) (AddGoalForm);
