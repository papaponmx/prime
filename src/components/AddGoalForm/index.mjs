import React from 'react'
import { Field, reduxForm } from 'redux-form'

const required = value => value ? undefined : 'Required'
const maxLength = (value) => value && value.length > 100 ? `Must be ${100} characters or less` : undefined
const minValue = (value) => value && value < 3 ? `Must be at least ${3}` : undefined

const renderField = ({ input, label, type, meta: { touched, error, warning } }) => (
  <div>
    <label>
      {label}
    </label>
    <div>
      <input {...input} placeholder={label} type={type} />
      {touched && ((error && <span>{error}</span>) || (warning && <span>{warning}</span>))}
    </div>
  </div>
)

const AddGoalForm = (props) => {
  const { handleSubmit, pristine, reset, submitting } = props
  return (
    <form onSubmit={handleSubmit}>
      <Field
        name='goal'
        type='text'
        component={renderField}
        label='Goal'
        validate={[required, maxLength, minValue]} />
      <div>
        <button type='submit' disabled={submitting}>
          Submit
        </button>
        <button type='button' disabled={pristine || submitting} onClick={reset}>
          Clear Values
        </button>
      </div>
    </form>
  )
}

export default reduxForm({
  form: 'addGoalForm'
})(AddGoalForm)
