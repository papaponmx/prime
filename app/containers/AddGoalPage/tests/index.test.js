import React from 'react';
import { shallow } from 'enzyme';

import { AddGoalPage } from '../index';
import AddGoalForm from '../AddGoalForm';

describe('<AddGoalPage />', () => {
  const component = shallow(<AddGoalPage />);
  const children = AddGoalForm;

  it('Should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should render a section', () => {
    expect(component.type()).toEqual('section');
  });

  it('Should have children', () => {
    expect(component.find(children)).toBeTruthy();
  });
});
