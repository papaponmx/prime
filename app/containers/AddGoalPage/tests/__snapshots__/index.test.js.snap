// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddGoalPage /> Should render 1`] = `
<section>
  <FormattedMessage
    defaultMessage="This is AddGoalPage container !"
    id="app.containers.AddGoalPage.header"
    values={Object {}}
  />
  <withReducer(withSaga(Connect(ReduxForm))) />
</section>
`;
